src/main/scala/mesosphere/marathon/MarathonScheduler.scala

```diff
@@ -29,32 +32,66 @@ class MarathonScheduler @Inject() (
     mesosLeaderInfo: MesosLeaderInfo,
     system: ActorSystem,
     config: MarathonConf,
-    schedulerCallbacks: SchedulerCallbacks) extends Scheduler {
+    schedulerCallbacks: SchedulerCallbacks,
+    @Named(ModuleNames.SCHEDULER_HEARTBEAT_MONITOR) heartbeatMonitor: ActorRef) extends Scheduler {
 
   private[this] val log = LoggerFactory.getLogger(getClass.getName)
 
   import scala.concurrent.ExecutionContext.Implicits.global
 
   implicit val zkTimeout = config.zkTimeoutDuration
 
+  // virtualHeartbeatTasks is sent in a reconciliation message to mesos in order to force a
+  // predictable response: the master (if we're connected) will send back a TASK_LOST because
+  // the fake task ID and agent ID that we use will never actually exist in the cluster.
+  // this is part of a short-term workaround: will no longer be needed once marathon is ported
+  // to use the new mesos v1 http API.
+  private[this] val virtualHeartbeatTasks: java.util.Collection[TaskStatus] = Seq(
+    TaskStatus.newBuilder
+    .setTaskId(TaskID.newBuilder.setValue("fake-marathon-pacemaker-task-" + UUID.randomUUID().toString))
+    .setState(TaskState.TASK_LOST) // required, so we just need to set something
+    .setSlaveId(SlaveID.newBuilder.setValue("fake-marathon-pacemaker-agent-" + UUID.randomUUID().toString))
+    .build
+  ).asJava
+
+  private[this] val heartbeatFailure = new Runnable {
```

Comment:
ack, the case class should just use higher order functions – yet I'm not sure whether the callbacks are actually needed (see my take on the Actor driving this on its own by calling the scheduler driver)